src/lib/ddl/event-handlers.js

```diff
@@ -0,0 +1,70 @@
+/* @flow */
+const noop = (e) => e === undefined;
+
+type EventHandlerType = {|
+  id : string,
+  predicate : (e : Object) => boolean,
+  handle : (e : Object) => void
+|};
+
+export const buildEventHandlers = (shoppingAnalytics : any) : $ReadOnlyArray<EventHandlerType> =>  {
+  const shoppingSendEventHandler : EventHandlerType = {
+    id: 'send_event_handler',
+    predicate: (e) => {
+      return e.event !== undefined;
+    },
+    handle: (e) => {
+      if (e.event !== 'snippetRun') {
+        shoppingAnalytics.send(e.event, e.payload);
+      }
+    }
+  };
+
+  const configurationEventHandler : EventHandlerType = {
+    id: 'set_handler',
+    predicate: (e) => {
+      return e.set !== undefined;
+    },
+    handle: (e) => {
+      shoppingAnalytics.set(e.set);
+    }
+  };
+
+  const autoGenerateProductPayloadEventHandler : EventHandlerType = {
+    id: 'autoGenerateProductPayload_handler',
+    predicate: (e) => {
+      return e.autoGenerateProductPayload !== undefined;
+    },
+    handle: (e) => {
+      shoppingAnalytics.autoGenerateProductPayload(e);
+    }
+  };
+  return [
+    shoppingSendEventHandler,
+    configurationEventHandler,
+    autoGenerateProductPayloadEventHandler
+  ];
+};
+
+export type EventConsumerType = {|
+  consume : (e : Object) => void
+|};
+
+export const createEventHandler = (shoppingAnalytics : any) : EventConsumerType => {
```

Comment:
The name is a but confusing since we already have a method called `buildEventHandlers`.
Can this be named to something like `createEventQueueProcessor` or ``createEventQueueConsumer` ?

Reply:
Sure, I'll create a separate PR to address code style comments